src/main/scala/edu/berkeley/ce/rockslicing/RockSlicer.scala

```diff
@@ -83,87 +81,59 @@ object RockSlicer {
-    // Search blocks for matching processor joints
-    if (processorBlocks.count() > 0) {
-      // Create pair RDD that contains initially all the blocks on each partion as the first entry and
-      // an empty sequence of blocks as the second entry. The first entry represents blocks that are yet
-      // to be merged, the second entry is blocks that have already been merged.
-      val treeReduceBlockPairsRDD = processorBlocks.map{ blocks => (Seq(blocks), Seq.empty[Block])}
-      val (allOrphanBlocks, allReconstructedBlocks) = treeReduceBlockPairsRDD.treeReduce({
-        case ((toMerge1, merged1), (toMerge2, merged2)) =>
-          // All blocks that share a processor joint are merged/reconstructed. Blocks that do not share
-          // a processor joint with any of the blocks that are to be merged are "orphan" blocks.
-          val (treeReconBlocks, treeOrphanBlocks) = mergeBlocks(toMerge1 ++ toMerge2, Seq.empty[Block],
-            Seq.empty[Block])
-
-          // Merged/reconstructed blocks are partitioned based on whether they still contain processor joints
-          val (treeProcessorBlocks, treeRealBlocks) = treeReconBlocks.partition { block =>
-            block.faces.exists(_.processorJoint)
-          }
-
-          // Orphan blocks and merged/reconstructed blocks that still contain processor joints are returned
-          // as the first entry to the pair RDD. These will be merged with the next level of the tree. Blocks
-          // that have no remaining processor joints are returned as the second entry in the pair RDD.
-          (treeProcessorBlocks ++ treeOrphanBlocks, treeRealBlocks ++ merged1 ++ merged2)
-        // The following formula specifies the tree depth for treeReduce. It ensures that at most two leaves are
-        // merged during each reduction.
-      }, math.ceil(math.log(arguments.numProcessors)/math.log(2)).toInt)
-
-      assert(allOrphanBlocks.isEmpty)
-
-      val reconCentroidBlocks = allReconstructedBlocks.map {block =>
-          val centroid = block.centroid
-          Block(centroid, block.updateFaces(centroid))
+    val mergedBlocks = if (processorBlocks.isEmpty) {
+      nonRedundantBlocks
+    } else {
+      /*
+       * Since we're dealing with RDDs and not Seqs here, we have to adapt
+       * the logic of LoadBalancer.mergeProcessorBlocks
+       */
+      val realBlocks = nonRedundantBlocks.filter { block =>
+        !block.faces.exists(_.isProcessorFace)
       }
 
-      // Clean up faces of reconstructed blocks with values that should be zero, but have
-      // arbitrarily small floating point values
-      val squeakyCleanRecon = reconCentroidBlocks.map { case Block(center, faces, generation) =>
-        Block(center, faces.map(_.applyTolerance), generation)
+      var orphanBlocks = processorBlocks.map { block =>
+        val globalOrigin = Array(0.0, 0.0, 0.0)
+        // Using same number of decimal places as in LoadBalancer for now
```

Comment:
We should talk about this some more and figure out what we're okay with regarding tolerance.